Branch-SDK/src/io/branch/indexing/ContentDiscoverer.java

```diff
@@ -151,83 +169,185 @@ public void run() {
+                                discoverContentKeys(rootView, contentKeysArray, activity.getResources());
                             }
                         }
                         discoveredViewList_.add(viewName);
-
+                        
                         // Cache the analytics data for future use
                         PrefHelper.getInstance(activity).saveBranchAnalyticsData(contentEvent_);
-
-                        //int discoveryRepeatTime = cdManifest_.getCDPathProperties(activity).getDiscoveryRepeatTime();
-                        //if (discoveryRepeatTime >= DRT_MINIMUM_THRESHHOLD) {
-                        //    handler_.postDelayed(readContentRunnable, discoveryRepeatTime);
-                        //} else {
-                        lastActivityReference_ = null;
-                        //}
+                        int discoveryRepeatTime = cdManifest_.getCDPathProperties(activity).getDiscoveryRepeatInterval();
+                        maxDiscoveryRepeatCnt = cdManifest_.getCDPathProperties(activity).getMaxDiscoveryRepeatNumber();
+                        if (discoveryRepeatCnt_ < maxDiscoveryRepeatCnt && discoveryRepeatTime >= DRT_MINIMUM_THRESHOLD && filteredElements != null && filteredElements.length() > 0) {
+                            handler_.postDelayed(readContentRunnable, discoveryRepeatTime);
+                        }
                     }
                 }
-
+                
             } catch (JSONException ignore) {
             }
         }
     };
-
-
-    private void discoverFilteredViewContents(JSONArray viewIDArray, JSONArray contentDataArray, JSONArray contentKeysArray, Activity activity, boolean isClearText) {
+    
+    private void discoverContentKeys(ViewGroup viewGroup, JSONArray contentKeysArray, Resources res) {
+        for (int i = 0; i < viewGroup.getChildCount(); i++) {
+            View childView = viewGroup.getChildAt(i);
+            if (childView.getVisibility() == View.VISIBLE) {
+                if (childView instanceof AbsListView || childView.getClass().getSimpleName().equals(RECYCLER_VIEW)) {
+                    discoverListViewContentKeys((ViewGroup) childView, res, contentKeysArray);
+                } else if (childView instanceof ViewGroup) {
+                    discoverContentKeys((ViewGroup) childView, contentKeysArray, res);
+                } else if (childView instanceof TextView) {
+                    String viewName = getViewName(childView, res);
+                    contentKeysArray.put(viewName);
+                }
+            }
+        }
+    }
+    
+    private void discoverListViewContentKeys(ViewGroup listView, Resources res, JSONArray contentKeysArray) {
+        JSONObject absListKeyObj = new JSONObject();
+        if (listView != null && listView.getChildCount() > -1) {
+            View candidateItemView = listView.getChildAt(listView.getChildCount() > 1 ? 1 : 0);
```

Comment:
can you comment on this logic? if the child count is > 1, you start at index 1? otherwise, if it's count =1 , use index 0?